lib/pages/new_group/new_group_chat_info.dart

```diff
@@ -1,429 +1,360 @@
+    if (currentPermissionPhotos != null && currentPermissionCamera != null) {
+      final imagePickerController = createImagePickerController();
+      groupNameFocusNode.unfocus();
+      showImagePickerBottomSheet(
+        context,
+        currentPermissionPhotos,
+        currentPermissionCamera,
+        imagePickerController,
+      );
+    }
+  }
 
-  const _AvatarForWebBuilder({
-    required this.avatarWebNotifier,
-  });
+  void _getDefaultGroupName(Set<PresentationContact> contactList) async {
+    if (contactList.length <= 3) {
+      final groupName =
+          contactList.map((contact) => contact.displayName).join(", ");
+      groupNameTextEditingController.text = groupName;
+      groupNameTextEditingController.selection = TextSelection.fromPosition(
+        TextPosition(offset: groupNameTextEditingController.text.length),
+      );
+      groupNameFocusNode.requestFocus();
+    } else {
+      groupNameTextEditingController.clear();
+    }
+  }
 
-  @override
-  Widget build(BuildContext context) {
-    return ValueListenableBuilder(
-      valueListenable: avatarWebNotifier,
-      builder: (context, value, child) {
-        if (value == null || value.files.single.bytes == null) {
-          return child!;
-        }
-        return ClipOval(
-          child: SizedBox.fromSize(
-            size:
-                const Size.fromRadius(NewGroupChatInfoStyle.avatarRadiusForWeb),
-            child: Image.memory(
-              value.files.single.bytes!,
-              fit: BoxFit.cover,
-              errorBuilder: (context, error, stackTrace) {
-                return const Center(
-                  child: Icon(Icons.error_outline),
-                );
-              },
-            ),
-          ),
-        );
-      },
-      child: Icon(
-        Icons.camera_alt_outlined,
-        color: Theme.of(context).colorScheme.surface,
-      ),
+  ImagePickerGridController createImagePickerController() {
+    final imagePickerController = ImagePickerGridController(
+      AssetCounter(imagePickerMode: ImagePickerMode.single),
     );
+
+    imagePickerController.addListener(() {
+      final selectedAsset = imagePickerController.selectedAssets.firstOrNull;
+      if (selectedAsset?.asset.type == AssetType.image) {
+        if (!imagePickerController.pickFromCamera()) {
+          Navigator.pop(context);
+        }
+        avatarAssetEntityNotifier.value = selectedAsset?.asset;
+        imagePickerController.removeAllSelectedItem();
+      }
+    });
+
+    return imagePickerController;
   }
-}
 
-class _EncryptionSettingTile extends StatelessWidget {
-  final ValueNotifier<bool> enableEncryptionNotifier;
+  bool get isCreatingRoom {
+    return createRoomStateNotifier.value.fold(
+          (failure) => false,
+          (success) =>
+              success is UploadContentLoading ||
+              success is CreateNewGroupChatLoading,
+        ) ??
+        false;
```

Comment:
why you need `?? false`

Reply:
i just moved the code out to the better class. But I have check it, actually no need `?? false`